src/jit/importer.cpp

```diff
@@ -1603,7 +1603,7 @@ GenTree* Compiler::impNormStructVal(GenTree*             structVal,
             break;
 
         case GT_IND:
-            assert(structVal->gtType == structType);
+            assert(varTypeIsStruct(structVal->gtType));
```

Comment:
I don't believe that this is correct. If this assert is failing, it means that `structVal` has not been properly normalized.